### PR TITLE
[ECO-519] Improved Logging

### PR DESCRIPTION
**Pull request by imonaco**
### What ?

Add more fields to the default logger

It resolves [this ticket](https://trustar.atlassian.net/jira/software/projects/ECO/boards/54?selectedIssue=ECO-519)

### Why ?

To improve observability of our lgos

### How ?

I add some optional fields, such as stash_config. They add some fields to the logging dict if are set. I also added env vars checking, defaulting to true, to have more control over what we want to log. Clean up some code also.

Example:

```json
{
  "message": "Started pulling data from 2021-07-14 21:26:59.368058+00:00 to 2021-07-14 21:33:37.306893+00:00",
  "level": "INFO",
  "module": "trustash.stashes.stash_base",
  "time": "2021-07-14T21:33:37.307654",
  "stash_type": "stash_crowdstrike",
  "stash_name": "trustar",
  "stash_id": "4bdc3f5b-3ed5-4d99-b20c-2d801866ef0b",
  "enclave_ids": "4bdc3f5b-3ed5-4d99-b20c-2d801866ef0b"
}
```

### CODE REVIEW

File: trustar2/log.py [hunk 1]
```diff
@@ -58,32 +94,55 @@ def _json_object_encoder(obj):
...
     Gets the handler to manage the output of the logger, default: stdout
     """
     handler = logging.StreamHandler(sys.stdout)
-    # TODO: read from a config file, or env var, and override the default formatter.
-    # IE: logging.FileHandler(filename='/path/to/file.log')
+    handler.setFormatter(formatter)
     return handler
 
+def get_file_handler(filename, formatter):
+    handler = logging.FileHandler(filename = filename)
```
Comment (manuelramos) [inline trustar2/log.py:106]: We should avoid use spaces around the = sign when used to indicate a keyword argument.

[Pep 8 Style Guide Reference](https://www.python.org/dev/peps/pep-0008/#other-recommendations)

File: trustar2/log.py [hunk 2]
```diff
@@ -58,32 +94,55 @@ def _json_object_encoder(obj):
...
-
-
-output_handler = get_handler()
-output_handler.setFormatter(get_formatter())
 
+def get_formatter(stash_config = None):
+    return TrustarJSONFormatter(stash_config = stash_config)
 
-def get_logger(name=None):
+def get_log_file(filename = None):
```
Comment (manuelramos) [inline trustar2/log.py:114]: Same comment as [here](https://github.com/trustar/trustar-sdk2-proto/pull/57/files#diff-5e3a4d6f9ad27a70ed6ebbbb152b8c9520a4b14f6aa27302f97c5aae90f88ed7R106)

File: trustar2/log.py [hunk 3]
```diff
@@ -58,32 +94,55 @@ def _json_object_encoder(obj):
...
+    with open(filename, "a+"):
+        return filename
+
+def get_logger(
+                name = None,
+                stash_config = None,
+                filename = None,
+                stdout_enabled = True,
+                level = None
+            ):
```
Comment (manuelramos) [inline trustar2/log.py:131]: Same as [here](https://github.com/trustar/trustar-sdk2-proto/pull/57/files#diff-5e3a4d6f9ad27a70ed6ebbbb152b8c9520a4b14f6aa27302f97c5aae90f88ed7R106)

File: trustar2/log.py [hunk 4]
```diff
@@ -58,32 +94,55 @@ def _json_object_encoder(obj):
...
     logger = logging.getLogger(name or __name__)
-    logger.addHandler(output_handler)
-    logger.setLevel(get_logging_level())
+
+    stdout_enabled = stdout_enabled and int(os.environ.get(LOGGING_STDOUT_ENABLED_VAR, 1))
+    if stdout_enabled:
+        stdout_formatter = get_formatter(stash_config = stash_config)
+        stdout_handler = get_stdout_handler( formatter = stdout_formatter)
```
Comment (manuelramos) [inline trustar2/log.py:137]: Same as [here](https://github.com/trustar/trustar-sdk2-proto/pull/57/files#diff-5e3a4d6f9ad27a70ed6ebbbb152b8c9520a4b14f6aa27302f97c5aae90f88ed7R106)

File: trustar2/log.py [hunk 5]
```diff
@@ -58,32 +94,55 @@ def _json_object_encoder(obj):
...
+        logger.addHandler(stdout_handler)
+    
+    file = get_log_file(filename)
+    if file:
+        file_formatter = get_formatter(stash_config = stash_config)
+        file_handler = get_file_handler(file, formatter = file_formatter)
```
Comment (manuelramos) [inline trustar2/log.py:143]: Same as [here](https://github.com/trustar/trustar-sdk2-proto/pull/57/files#diff-5e3a4d6f9ad27a70ed6ebbbb152b8c9520a4b14f6aa27302f97c5aae90f88ed7R106)

**Review by manuelramos — APPROVED** [review]
Looks good!
Just leave an style-related comment, but I'm approving this PR anyway.
Thanks Ivan!

File: trustar2/log.py (hunk 1 above)
Comment (imonaco) [reply trustar2/log.py:106]: Oh, I thought that good practice was to leave the space... Thanks!

**Review by hernantain — APPROVED** [review]
LGTM!

**Comment by marcos-vivar** [comment]
@imonaco outstanding refactor! 👍🏽

**Pull request merged**